dataland-frontend/src/components/general/UserProfileDropDown.vue

```diff
@@ -55,40 +73,38 @@ export default {
   },
 
   methods: {
-    toggleDropdownMenu(event) {
-      this.$refs.menu.toggle(event);
-    },
     logoutViaDropdown() {
-      this.getKeycloakPromise()
+      this.getKeycloakPromise?.()
         .then((keycloak) => {
           if (keycloak.authenticated) {
-            let baseUrl = window.location.origin;
+            const baseUrl = window.location.origin;
             const url = keycloak.createLogoutUrl({ redirectUri: `${baseUrl}` });
             location.assign(url);
           }
         })
-        .catch((error) => console.log("error: " + error));
+        .catch((error) => console.log(error));
     },
     gotoUserSettings() {
-      this.getKeycloakPromise()
-        .then((keycloak) => {
+      this.getKeycloakPromise?.()
+        .then(async (keycloak) => {
```

Comment:
Brauchen wir das async/await hier? Wird das nicht durch das then von alleine gemacht? 
Ah - vermutlich weil das in der klammer eine void funktion ist. 
Wie wäre es damit, statt async/await in dem Methodenblock ein return zu verwenden? 
Also:
```
      this.getKeycloakPromise?.()
        .then((keycloak) => {
          if (keycloak.authenticated) {
            return keycloak.accountManagement();
          }
        })
        .catch((error) => console.log(error));

```

Reply:
finde ich gut, damit wird die Intention auch klarer

Reply:
ok